src/contracts/pods/EigenPod.sol

```diff
@@ -164,267 +175,355 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-     * @param validatorFieldsProof is the information needed to check the veracity of the validator fields being proven
+     * @param validatorFieldsProofs is the information needed to check the veracity of the validator fields being proven
      * @param withdrawalFields are the fields of the withdrawal being proven
      * @param validatorFields are the fields of the validator being proven
      * @param beaconChainETHStrategyIndex is the index of the beaconChainETHStrategy for the pod owner for the callback to 
      *        the EigenPodManager to the StrategyManager in case it must be removed from the podOwner's list of strategies
-     * @param oracleBlockNumber is the Beacon Chain blockNumber whose state root the `proof` will be proven against.
+     * @param oracleTimestamp is the Beacon Chain blockNumber whose state root the `proof` will be proven against.
      */
-    function verifyAndProcessWithdrawal(
+    function verifyAndProcessWithdrawals(
+        BeaconChainProofs.WithdrawalProofs[] calldata withdrawalProofs, 
+        bytes[] calldata validatorFieldsProofs,
+        bytes32[][] calldata validatorFields,
+        bytes32[][] calldata withdrawalFields,
+        uint256 beaconChainETHStrategyIndex,
+        uint64 oracleTimestamp
+    )
+        external
+        onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_WITHDRAWAL)
+        onlyNotFrozen
+    {
+        require(
+            (validatorFields.length == validatorFieldsProofs.length) && 
+            (validatorFieldsProofs.length == withdrawalProofs.length) && 
+            (withdrawalProofs.length == withdrawalFields.length), "EigenPod.verifyAndProcessWithdrawals: inputs must be same length"
+        );
+
+        for (uint256 i = 0; i < withdrawalFields.length; i++) {
+            _verifyAndProcessWithdrawal(withdrawalProofs[i], validatorFieldsProofs[i], validatorFields[i], withdrawalFields[i], beaconChainETHStrategyIndex, oracleTimestamp);
+        }
+    }
+
+    function decrementWithdrawableRestakedExecutionLayerGwei(uint256 amountWei) external onlyEigenPodManager {
+        uint64 amountGwei = uint64(amountWei / GWEI_TO_WEI);
+        require(withdrawableRestakedExecutionLayerGwei >= amountGwei , "EigenPod.decrementWithdrawableRestakedExecutionLayerGwei: amount to decrement is greater than current withdrawableRestakedRxecutionLayerGwei balance");
+        withdrawableRestakedExecutionLayerGwei -= amountGwei;
+    }
+
+    function incrementWithdrawableRestakedExecutionLayerGwei(uint256 amountWei) external onlyEigenPodManager {
+        uint64 amountGwei = uint64(amountWei / GWEI_TO_WEI);
+        withdrawableRestakedExecutionLayerGwei += amountGwei;
+    }
+
+    function _verifyWithdrawalCredentials(
+        uint64 oracleTimestamp,
+        uint40 validatorIndex,
+        BeaconChainProofs.WithdrawalCredentialProofs calldata proofs,
+        bytes32[] calldata validatorFields
+    )
+        internal
+        returns (uint256)
+    {
```

Comment:
pretty significant function, would help to have a top-level explanation at least